roles/satellite-clone/tasks/restore.yml

```diff
@@ -7,23 +7,26 @@
 
 - name: Restore postgresql data
   command: tar --selinux --overwrite -xvf {{ backup_dir }}/pgsql_data.tar.gz -C /
-  when: not rhel_migration
+  when: not rhel_migration and pgsql_data.stat.exists
```

Comment:
The rhel_migration check is unnecessary. Check for pgsql_data.stat.exists is unnecessary as the validation steps are already performed.

This should just say:
when: standard_backup